examples/platform/esp32/time/TimeSync.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "TimeSync.h"
+#include <esp_sntp.h>
+#include <lib/support/CHIPMemString.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#define REF_TIME 1546300800 /* 01-Jan-2019 00:00:00 */
```

Comment:
maybe this snould be named `REF_TIME_EPOCH_SEC`

could we use `static constexpr` for these to be typesafe? like:

```cpp
static constexpr time_t kMinValidTimeStampEpoch = 1546300800; // 1 Jan 2019
static constexpr uint32_t kSecondsInADay = 24*60*60;
```

Reply:
BTW, the earliest time possible for SNTP should definitely be > Jan 1 2024...

Reply:
Isn't Matter EPOCH defined as the "1st of January, 2000 UTC (the Epoch)" (See main spec chapter 7.19.2.5).
If this is a reference to some arbitrary time would be good to not to call it Epoch, or to use 1st Jan 2000.

Reply:
Used `Jan 1 2024` as reference time